includes/Core/Permissions/Permissions.php

```diff
@@ -399,6 +414,46 @@ private function check_dashboard_sharing_capability( $cap, $user_id, $args ) {
 		}
 	}
 
+	/**
+	 * Checks if the VIEW_SPLASH capability is allowed for the user.
+	 *
+	 * Allows access to the VIEW_SPLASH capability if the user can authenticate or has
+	 * VIEW_SHARED_DASHBOARD capability.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param int $user_id User ID of the user the capability is checked for.
+	 * @return array Array with a 'do_not_allow' element if checks fail, empty array if checks pass.
+	 */
+	private function check_view_splash_capability( $user_id ) {
+		if ( Feature_Flags::enabled( 'dashboardSharing' ) ) {
+			if ( in_array( 'do_not_allow', $this->check_view_shared_dashboard_capability( $user_id ), true ) ) {
+				return array( self::AUTHENTICATE );
+			}
+			return array();
+		}
+		return array( self::AUTHENTICATE );
+	}
+
+	/**
+	 * Checks if the VIEW_DASHBOARD capability is allowed for the user.
+	 *
+	 * Allows access to the VIEW_DASHBOARD capability if the user can view either
+	 * the authenticated or shared dashboard.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param int $user_id User ID of the user the capability is checked for.
+	 * @return array Array with a 'do_not_allow' element if checks fail, empty array if checks pass.
+	 */
+	private function check_view_dashboard_capability( $user_id ) {
+		if ( in_array( 'do_not_allow', $this->check_view_authenticated_dashboard_capability( $user_id ), true ) ) {
+			return $this->check_view_shared_dashboard_capability( $user_id );
```

Comment:
Technically, we only need to call `check_view_shared_dashboard_capability` here if `dashboardSharing` is enabled, since otherwise there is no shared dashboard. Without the feature flag, we should technically return the result of `check_view_authenticated_dashboard_capability` here.

Basically, without the `dashboardSharing` flag, the `check_view_dashboard_capability` method should return exactly the same as `check_view_authenticated_dashboard_capability`.

This isn't really critical, since both `VIEW_DASHBOARD` and `VIEW_POSTS_INSIGHTS` are already covered by the more general condition in `map_meta_cap`, but probably a bit cleaner to make this method work for the current (without feature flag) requirements as well, just in case.